cmd/containerd-shim-runhcs-v1/task_hcs.go

```diff
@@ -578,8 +578,9 @@ func (ht *hcsTask) DeleteExec(ctx context.Context, eid string) (int, uint32, tim
 				ex.ForceExit(ctx, 1)
 			}
 
-			// iterate next
-			return false
+			// iterate next. Returning false here, means only sending
+			// Kill to the first task in the map.
+			return true
```

Comment:
There are several places (line 540 in `KillExec` and 725 in `waitForHostExit`) where the comment says iterate all but returns false. I wonder if there was an API change. Or this is just a mistake we never caught

Reply:
execs seems to be a ```sync.Map```. The docs for Range() states that f() returns false, iteration stops:

https://pkg.go.dev/sync#Map.Range

Reply:
Going back in time, it looks like that's been the case since 2017; and we have several bugs here where we aren't correctly iterating over all the execs.
Thats a good catch

Reply:
I took the liberty of fixing those returns as well. Updated the PR.